x/deployment/keeper/keeper.go

```diff
@@ -84,6 +83,28 @@ func (k Keeper) GetGroups(ctx sdk.Context, id types.DeploymentID) []types.Group
 	return vals
 }
 
+// GetOpenGroups returns all *open* groups of a deployment with given DeploymentID.
+//
+func (k Keeper) GetOpenGroups(ctx sdk.Context, id types.DeploymentID) []types.Group {
+	store := ctx.KVStore(k.skey)
+	key := groupsOpenKey(id)
+
+	var vals []types.Group
+	iter := sdk.KVStorePrefixIterator(store, key)
+	for ; iter.Valid(); iter.Next() {
+		gkey, _ := groupOpenKeyConvert(iter.Key())
+
+		buf := store.Get(gkey)
+		var val types.Group
+		k.cdc.MustUnmarshalBinaryBare(buf, &val)
+		if err := val.ValidateOrderable(); err == nil {
```

Comment:
I agree it should never happen, that's why I figured it'd be best to skip adding it to the return list if it had that state. My inclination is to avoid calling `panic` if there are recovery mechanisms in place.

However this function is currently only used for validation in tests at the moment. I figured it might be a useful utility function down the road, and if it lives in a mission critical case, then maybe we revisit if this should actually panic.

Reply:
i see.  if it's for tests, just make a helper.  it can use the index or not, it's up to you and the test. maybe later we can revisit, but this is a critical change & i don't want to add unnecessary confusion.

Reply:
Okay, reduced it to an internal function and added other state checks using GetGroups() method in tests.